There are failing tests when CERTIFICATE_LINKEDIN setting is configured at common.py

In order to be able to share course certificate to LinkedIn, an operator has to configure SOCIAL_SHARING_SETTINGS with a value True for CERTIFICATE_LINKEDIN. The setting is not available by default.

The following configuration will enable Sharing to LinkedIn on lms.
```
############################# SOCIAL MEDIA SHARING #############################
SOCIAL_SHARING_SETTINGS = {
    # Note: Ensure 'CUSTOM_COURSE_URLS' has a matching value in lms/envs/common.py
    'CUSTOM_COURSE_URLS': False,
    'DASHBOARD_FACEBOOK': False,
    'CERTIFICATE_FACEBOOK': False,
    'CERTIFICATE_TWITTER': False,
    'DASHBOARD_TWITTER': False,
    'CERTIFICATE_LINKEDIN': True
}

```
### The problem
Once the setting is configured as above(set either True/False), you see there are failing tests.

### To Reproduce
Steps to reproduce the behavior:
1. Configure common.py as bellow

```
############################# SOCIAL MEDIA SHARING #############################
SOCIAL_SHARING_SETTINGS = {
    # Note: Ensure 'CUSTOM_COURSE_URLS' has a matching value in lms/envs/common.py
    'CUSTOM_COURSE_URLS': False,
    'DASHBOARD_FACEBOOK': False,
    'CERTIFICATE_FACEBOOK': False,
    'CERTIFICATE_TWITTER': False,
    'DASHBOARD_TWITTER': False,
    'CERTIFICATE_LINKEDIN': False
}
```

2. Run the following tests

`app@6ab2e0cce482:~/edx-platform$ python -Wd -m pytest -p no:randomly --ds=lms.envs.test ./common/djangoapps/student/tests/
`


3. See failing tests
When set to False, you will see the following failing tests:

```
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
==================================================================================================================== slowest 20 durations ====================================================================================================================
6.72s setup    common/djangoapps/student/tests/test_activate_account.py::TestActivateAccount::test_account_activation_invalid_next_url_redirects_dashboard
2.74s call     common/djangoapps/student/tests/test_views.py::StudentDashboardTests::test_dashboard_with_resume_buttons_and_view_buttons
1.81s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_enroll_in_proctored_course_1___honor___False_
1.70s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_enroll_1___________audit__
1.69s call     common/djangoapps/student/tests/test_views.py::StudentDashboardTests::test_resume_course_appears_on_dashboard
1.63s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_with_invalid_course_id
1.59s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_upgrade_proctoring_enrollment_3_professional
1.50s call     common/djangoapps/student/tests/test_email.py::EmailChangeConfirmationTests::test_new_email_fails
1.44s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_score_recalculation_on_enrollment_update
1.43s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_enroll_in_proctored_course_7___credit___False_
1.43s call     common/djangoapps/student/tests/test_admin_views.py::TestUserProfileAutocompleteAdmin::test_admin_inline_autocomplete_urls_render
1.41s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_enroll_in_proctored_course_3___verified___True_
1.40s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_enroll_3____honor____verified____audit_____course_modes_choose____honor__
1.39s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_unenroll_when_unenrollment_disabled
1.36s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_enroll_with_email_opt_in_10____no_id_professional_____true__
1.35s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_enroll_with_email_opt_in_02_______false__
1.34s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_enroll_with_email_opt_in_05____honor____verified_____false__
1.33s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_enroll_in_proctored_course_5___professional___True_
1.33s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_enroll_with_email_opt_in_07____professional_____true__
1.31s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_invalid_enrollment_action
================================================================================================================== short test summary info ===================================================================================================================
FAILED common/djangoapps/student/tests/test_certificates.py::CertificateDisplayTest::test_post_to_linkedin_visibility - AssertionError: False is not true : Couldn't find 'Add Certificate to LinkedIn Profile' in response
FAILED common/djangoapps/student/tests/tests.py::DashboardTest::test_linked_in_add_to_profile_btn_with_certificate - AssertionError: False is not true : Couldn't find 'Add Certificate to LinkedIn' in response
============================================================================================ 2 failed, 583 passed, 6 skipped, 79651 warnings in 154.15s (0:02:34) ============================================================================================
app@6ab2e0cce482:~/edx-platform$
```


4. Change configuration as bellow
```
############################# SOCIAL MEDIA SHARING #############################
SOCIAL_SHARING_SETTINGS = {
    # Note: Ensure 'CUSTOM_COURSE_URLS' has a matching value in lms/envs/common.py
    'CUSTOM_COURSE_URLS': False,
    'DASHBOARD_FACEBOOK': False,
    'CERTIFICATE_FACEBOOK': False,
    'CERTIFICATE_TWITTER': False,
    'DASHBOARD_TWITTER': False,
    'CERTIFICATE_LINKEDIN': True
}
```


5. Run the following tests

`app@6ab2e0cce482:~/edx-platform$ python -Wd -m pytest -p no:randomly --ds=lms.envs.test ./common/djangoapps/student/tests/
`


6. See failing tests
When set to True, you will see the following failing tests:


```
-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
==================================================================================================================== slowest 20 durations ====================================================================================================================
6.45s setup    common/djangoapps/student/tests/test_activate_account.py::TestActivateAccount::test_account_activation_invalid_next_url_redirects_dashboard
2.56s call     common/djangoapps/student/tests/test_views.py::StudentDashboardTests::test_dashboard_with_resume_buttons_and_view_buttons
1.56s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_upgrade_proctoring_enrollment_4_executive_education
1.50s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_enroll_1___________audit__
1.48s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_cea_enrolls_only_one_user
1.43s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_upgrade_proctoring_enrollment_2_masters
1.41s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_enroll_in_proctored_course_7___credit___False_
1.41s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_unenroll_when_unenrollment_disabled
1.41s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_enroll_in_proctored_course_5___professional___True_
1.40s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_enroll_with_email_opt_in_12____no_id_professional____None_
1.39s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_score_recalculation_on_enrollment_update
1.39s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_user_not_authenticated
1.37s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_enroll_with_email_opt_in_04____honor____verified_____true__
1.37s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_enroll_3____honor____verified____audit_____course_modes_choose____honor__
1.35s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_external_course_updates_signal_1__1
1.35s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_external_course_updates_signal_3_1
1.32s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_enroll_with_email_opt_in_07____professional_____true__
1.32s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_enroll_with_email_opt_in_10____no_id_professional_____true__
1.31s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_enroll_in_proctored_course_3___verified___True_
1.30s call     common/djangoapps/student/tests/test_enrollment.py::EnrollmentTest::test_enroll_with_email_opt_in_02_______false__
================================================================================================================== short test summary info ===================================================================================================================
FAILED common/djangoapps/student/tests/test_certificates.py::CertificateDisplayTest::test_post_to_linkedin_visibility - AssertionError: 1 != 0 : Response should not contain 'Add Certificate to LinkedIn Profile'
FAILED common/djangoapps/student/tests/tests.py::CourseEndingTest::test_cert_info - AssertionError: assert {'can_unenrol...C3%A9dX', ...} == {'can_unenrol...l': None, ...}
FAILED common/djangoapps/student/tests/tests.py::CourseEndingTest::test_cert_info_beta_tester - AssertionError: assert {'can_unenrol...C3%A9dX', ...} == {'can_unenrol...l': None, ...}
FAILED common/djangoapps/student/tests/tests.py::DashboardTest::test_linked_in_add_to_profile_btn_not_appearing_without_config - AssertionError: 1 != 0 : Response should not contain 'Add Certificate to LinkedIn'
============================================================================================ 4 failed, 581 passed, 6 skipped, 79651 warnings in 139.06s (0:02:19) ============================================================================================
app@6ab2e0cce482:~/edx-platform$ 
```


### Expected Behavior
Tests, regardless of configuration settings should be able to test functionalities by mocking inputs or overriding setting information independently.

--- a/common/djangoapps/student/tests/test_certificates.py
+++ b/common/djangoapps/student/tests/test_certificates.py
@@ -17,10 +17,8 @@ from common.djangoapps.student.tests.factories import CourseEnrollmentFactory, U
 from openedx.core.djangolib.testing.utils import skip_unless_lms
 from lms.djangoapps.certificates.api import get_certificate_url
 from lms.djangoapps.certificates.data import CertificateStatuses
-from lms.djangoapps.certificates.tests.factories import (
-    GeneratedCertificateFactory,
-    LinkedInAddToProfileConfigurationFactory
-)
+from lms.djangoapps.certificates.tests.factories import GeneratedCertificateFactory
+from openedx.core.djangoapps.site_configuration.tests.test_util import with_site_configuration_context
 
 # pylint: disable=no-member
 
@@ -155,22 +153,29 @@ class CertificateDisplayTest(CertificateDisplayTestBase):
             'do not have a current verified identity with {platform_name}'
             .format(platform_name=settings.PLATFORM_NAME))
 
-    def test_post_to_linkedin_visibility(self):
+    @ddt.data(
+        (True, True),
+        (False, False),
+    )
+    @ddt.unpack
+    def test_post_to_linkedin_visibility(self, certificate_linkedin_enabled, linkedin_button_visible):
         """
-        Verifies that the post certificate to linked button
-        does not appear by default (when config is not set)
-        Then Verifies that the post certificate to linked button appears
-        as expected once a config is set
+        Verify the LinkedIn "Add to Profile" button visibility based on configuration.
+
+        Tests that:
+        1. When CERTIFICATE_LINKEDIN is False, the LinkedIn button is not visible
+        2. When CERTIFICATE_LINKEDIN is True, the LinkedIn button appears as expected
         """
         self._create_certificate('honor')
 
-        # until we set up the configuration, the LinkedIn action
-        # button should not be visible
-        self._check_linkedin_visibility(False)
-
-        LinkedInAddToProfileConfigurationFactory()
-        # now we should see it
-        self._check_linkedin_visibility(True)
+        # LinkedIn sharing Status True/False
+        # When CERTIFICATE_LINKEDIN is set to False in site configuration,
+        # the LinkedIn "Add to Profile" button should not be visible to users
+        # but if set to True the LinkedIn "Add to Profile" button should be visible
+        # to users allowing them to share their certificate on LinkedIn
+        SITE_CONFIGURATION = {"SOCIAL_SHARING_SETTINGS": {"CERTIFICATE_LINKEDIN": certificate_linkedin_enabled}}
+        with with_site_configuration_context(configuration=SITE_CONFIGURATION):
+            self._check_linkedin_visibility(linkedin_button_visible)
 
 
 @ddt.ddt

--- a/common/djangoapps/student/tests/tests.py
+++ b/common/djangoapps/student/tests/tests.py
@@ -51,6 +51,8 @@ from openedx.features.course_experience.url_helpers import make_learning_mfe_cou
 from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase, SharedModuleStoreTestCase  # lint-amnesty, pylint: disable=wrong-import-order
 from xmodule.modulestore.tests.factories import CourseFactory, check_mongo_calls  # lint-amnesty, pylint: disable=wrong-import-order
 from xmodule.data import CertificatesDisplayBehaviors  # lint-amnesty, pylint: disable=wrong-import-order
+from openedx.core.djangoapps.site_configuration.tests.test_util import with_site_configuration_context
+
 log = logging.getLogger(__name__)
 
 BETA_TESTER_METHOD = 'common.djangoapps.student.helpers.access.is_beta_tester'
@@ -90,85 +92,132 @@ class CourseEndingTest(ModuleStoreTestCase):
             download_url='http://s3.edx/cert'
         )
         enrollment = CourseEnrollmentFactory(user=user, course_id=course.id, mode=CourseMode.VERIFIED)
-
-        assert _cert_info(user, enrollment, None) ==\
-               {'status': 'processing', 'show_survey_button': False, 'can_unenroll': True}
-
-        cert_status = {'status': 'unavailable', 'mode': 'honor', 'uuid': None}
-        assert _cert_info(user, enrollment, cert_status) == {'status': 'processing', 'show_survey_button': False,
-                                                             'mode': 'honor', 'linked_in_url': None,
-                                                             'can_unenroll': True}
-
-        cert_status = {'status': 'generating', 'grade': '0.67', 'mode': 'honor', 'uuid': None}
-        with patch('lms.djangoapps.grades.course_grade_factory.CourseGradeFactory.read') as patch_persisted_grade:
-            patch_persisted_grade.return_value = Mock(percent=1.0)
-            assert _cert_info(user, enrollment, cert_status) == {'status': 'generating', 'show_survey_button': True,
-                                                                 'survey_url': survey_url, 'grade': '1.0',
-                                                                 'mode': 'honor', 'linked_in_url': None,
-                                                                 'can_unenroll': False}
-
-        cert_status = {'status': 'generating', 'grade': '0.67', 'mode': 'honor', 'uuid': None}
-        assert _cert_info(user, enrollment, cert_status) == {'status': 'generating', 'show_survey_button': True,
-                                                             'survey_url': survey_url, 'grade': '0.67', 'mode': 'honor',
-                                                             'linked_in_url': None, 'can_unenroll': False}
-
-        cert_status = {
-            'status': 'downloadable',
-            'grade': '0.67',
-            'download_url': cert.download_url,
-            'mode': 'honor',
-            'uuid': 'fakeuuidbutitsfine',
+        # LinkedIn sharing disabled as we are expecting 'linked_in_url': None in our test case
+        SITE_CONFIGURATION = {
+            'SOCIAL_SHARING_SETTINGS': {
+                'CERTIFICATE_LINKEDIN': False
+            }
         }
-        assert _cert_info(user, enrollment, cert_status) == {'status': 'downloadable',
-                                                             'download_url': cert.download_url,
-                                                             'show_survey_button': True, 'survey_url': survey_url,
-                                                             'grade': '0.67', 'mode': 'honor', 'linked_in_url': None,
-                                                             'can_unenroll': False}
+        with with_site_configuration_context(configuration=SITE_CONFIGURATION):
+            assert _cert_info(user, enrollment, None) == {
+                'status': 'processing',
+                'show_survey_button': False,
+                'can_unenroll': True,
+            }
 
-        cert_status = {
-            'status': 'notpassing', 'grade': '0.67',
-            'download_url': cert.download_url,
-            'mode': 'honor',
-            'uuid': 'fakeuuidbutitsfine',
-        }
-        assert _cert_info(user, enrollment, cert_status) == {'status': 'notpassing', 'show_survey_button': True,
-                                                             'survey_url': survey_url, 'grade': '0.67', 'mode': 'honor',
-                                                             'linked_in_url': None, 'can_unenroll': True}
+            cert_status = {'status': 'unavailable', 'mode': 'honor', 'uuid': None}
+            assert _cert_info(user, enrollment, cert_status) == {
+                'status': 'processing',
+                'show_survey_button': False,
+                'mode': 'honor',
+                'linked_in_url': None,
+                'can_unenroll': True,
+            }
 
-        # Test a course that doesn't have a survey specified
-        course2 = CourseOverviewFactory.create(
-            end_of_course_survey_url=None,
-            certificates_display_behavior='end',
-        )
-        enrollment2 = CourseEnrollmentFactory(user=user, course_id=course2.id, mode=CourseMode.VERIFIED)
+            cert_status = {'status': 'generating', 'grade': '0.67', 'mode': 'honor', 'uuid': None}
+            with patch('lms.djangoapps.grades.course_grade_factory.CourseGradeFactory.read') as patch_persisted_grade:
+                patch_persisted_grade.return_value = Mock(percent=1.0)
+                assert _cert_info(user, enrollment, cert_status) == {
+                    'status': 'generating',
+                    'show_survey_button': True,
+                    'survey_url': survey_url,
+                    'grade': '1.0',
+                    'mode': 'honor',
+                    'linked_in_url': None,
+                    'can_unenroll': False,
+                }
 
-        cert_status = {
-            'status': 'notpassing', 'grade': '0.67',
-            'download_url': cert.download_url, 'mode': 'honor', 'uuid': 'fakeuuidbutitsfine'
-        }
-        assert _cert_info(user, enrollment2, cert_status) == {'status': 'notpassing', 'show_survey_button': False,
-                                                              'grade': '0.67', 'mode': 'honor', 'linked_in_url': None,
-                                                              'can_unenroll': True}
+            cert_status = {'status': 'generating', 'grade': '0.67', 'mode': 'honor', 'uuid': None}
+            assert _cert_info(user, enrollment, cert_status) == {
+                'status': 'generating',
+                'show_survey_button': True,
+                'survey_url': survey_url,
+                'grade': '0.67',
+                'mode': 'honor',
+                'linked_in_url': None,
+                'can_unenroll': False,
+            }
 
-        course3 = CourseOverviewFactory.create(
-            end_of_course_survey_url=None,
-            certificates_display_behavior='early_no_info',
-        )
-        enrollment3 = CourseEnrollmentFactory(user=user, course_id=course3.id, mode=CourseMode.VERIFIED)
-        # test when the display is unavailable or notpassing, we get the correct results out
-        course2.certificates_display_behavior = CertificatesDisplayBehaviors.EARLY_NO_INFO
-        cert_status = {'status': 'unavailable', 'mode': 'honor', 'uuid': None}
-        assert _cert_info(user, enrollment3, cert_status) == {'status': 'processing', 'show_survey_button': False,
-                                                              'can_unenroll': True}
+            cert_status = {
+                'status': 'downloadable',
+                'grade': '0.67',
+                'download_url': cert.download_url,
+                'mode': 'honor',
+                'uuid': 'fakeuuidbutitsfine',
+            }
+            assert _cert_info(user, enrollment, cert_status) == {
+                'status': 'downloadable',
+                'download_url': cert.download_url,
+                'show_survey_button': True,
+                'survey_url': survey_url,
+                'grade': '0.67',
+                'mode': 'honor',
+                'linked_in_url': None,
+                'can_unenroll': False,
+            }
 
-        cert_status = {
-            'status': 'notpassing', 'grade': '0.67',
-            'download_url': cert.download_url,
-            'mode': 'honor',
-            'uuid': 'fakeuuidbutitsfine'
-        }
-        assert _cert_info(user, enrollment3, cert_status) == {'status': 'processing', 'show_survey_button': False,
-                                                              'can_unenroll': True}
+            cert_status = {
+                'status': 'notpassing',
+                'grade': '0.67',
+                'download_url': cert.download_url,
+                'mode': 'honor',
+                'uuid': 'fakeuuidbutitsfine',
+            }
+            assert _cert_info(user, enrollment, cert_status) == {
+                'status': 'notpassing',
+                'show_survey_button': True,
+                'survey_url': survey_url,
+                'grade': '0.67',
+                'mode': 'honor',
+                'linked_in_url': None,
+                'can_unenroll': True,
+            }
+
+            # Test a course that doesn't have a survey specified
+            course2 = CourseOverviewFactory.create(
+                end_of_course_survey_url=None,
+                certificates_display_behavior='end',
+            )
+            enrollment2 = CourseEnrollmentFactory(user=user, course_id=course2.id, mode=CourseMode.VERIFIED)
+
+            cert_status = {
+                'status': 'notpassing', 'grade': '0.67',
+                'download_url': cert.download_url, 'mode': 'honor', 'uuid': 'fakeuuidbutitsfine'
+            }
+            assert _cert_info(user, enrollment2, cert_status) == {
+                'status': 'notpassing',
+                'show_survey_button': False,
+                'grade': '0.67',
+                'mode': 'honor',
+                'linked_in_url': None,
+                'can_unenroll': True
+            }
+
+            course3 = CourseOverviewFactory.create(
+                end_of_course_survey_url=None,
+                certificates_display_behavior='early_no_info',
+            )
+            enrollment3 = CourseEnrollmentFactory(user=user, course_id=course3.id, mode=CourseMode.VERIFIED)
+            # test when the display is unavailable or notpassing, we get the correct results out
+            course2.certificates_display_behavior = CertificatesDisplayBehaviors.EARLY_NO_INFO
+            cert_status = {'status': 'unavailable', 'mode': 'honor', 'uuid': None}
+            assert _cert_info(user, enrollment3, cert_status) == {
+                'status': 'processing',
+                'show_survey_button': False,
+                'can_unenroll': True
+            }
+
+            cert_status = {
+                'status': 'notpassing', 'grade': '0.67',
+                'download_url': cert.download_url,
+                'mode': 'honor',
+                'uuid': 'fakeuuidbutitsfine'
+            }
+            assert _cert_info(user, enrollment3, cert_status) == {
+                'status': 'processing',
+                'show_survey_button': False,
+                'can_unenroll': True
+            }
 
     def test_cert_info_beta_tester(self):
         user = UserFactory.create()
@@ -192,15 +241,22 @@ class CourseEndingTest(ModuleStoreTestCase):
             'uuid': 'blah',
         }
         with patch(BETA_TESTER_METHOD, return_value=False):
-            assert _cert_info(user, enrollment, cert_status) == {
-                'status': status,
-                'download_url': cert.download_url,
-                'show_survey_button': False,
-                'grade': grade,
-                'mode': mode,
-                'linked_in_url': None,
-                'can_unenroll': False
+            # LinkedIn sharing disabled as we are expecting 'linked_in_url': None in our test case
+            SITE_CONFIGURATION = {
+                'SOCIAL_SHARING_SETTINGS': {
+                    'CERTIFICATE_LINKEDIN': False
+                }
             }
+            with with_site_configuration_context(configuration=SITE_CONFIGURATION):
+                assert _cert_info(user, enrollment, cert_status) == {
+                    'status': status,
+                    'download_url': cert.download_url,
+                    'show_survey_button': False,
+                    'grade': grade,
+                    'mode': mode,
+                    'linked_in_url': None,
+                    'can_unenroll': False
+                }
 
         with patch(BETA_TESTER_METHOD, return_value=True):
             assert _cert_info(user, enrollment, cert_status) == {
@@ -397,7 +453,16 @@ class DashboardTest(ModuleStoreTestCase, TestVerificationBase):
             grade='67',
             download_url=download_url
         )
-        response = self.client.get(reverse('dashboard'))
+        # LinkedIn sharing disabled
+        # When CERTIFICATE_LINKEDIN is set to False in site configuration,
+        # the LinkedIn "Add to Profile" button should not be visible to users
+        SITE_CONFIGURATION = {
+            'SOCIAL_SHARING_SETTINGS': {
+                'CERTIFICATE_LINKEDIN': False
+            }
+        }
+        with with_site_configuration_context(configuration=SITE_CONFIGURATION):
+            response = self.client.get(reverse('dashboard'))
 
         assert response.status_code == 200
         self.assertNotContains(response, 'Add Certificate to LinkedIn')
@@ -435,7 +500,16 @@ class DashboardTest(ModuleStoreTestCase, TestVerificationBase):
             grade='67',
             download_url='https://www.edx.org'
         )
-        response = self.client.get(reverse('dashboard'))
+        # LinkedIn sharing disabled
+        # When CERTIFICATE_LINKEDIN is set to False in site configuration,
+        # the LinkedIn "Add to Profile" button should not be visible to users
+        SITE_CONFIGURATION = {
+            'SOCIAL_SHARING_SETTINGS': {
+                'CERTIFICATE_LINKEDIN': True
+            }
+        }
+        with with_site_configuration_context(configuration=SITE_CONFIGURATION):
+            response = self.client.get(reverse('dashboard'))
 
         assert response.status_code == 200
         self.assertContains(response, 'Add Certificate to LinkedIn')

